src/map/lua/lua_baseentity.cpp

```diff
@@ -1878,6 +1996,100 @@ inline int32 CLuaBaseEntity::isFollowingPath(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+* Function: rotateToAngle()
+* Purpose : rotates an entity to a given rotation
+* Example : npc:rotateToAngle(180)
+* Notes   : 
+************************************************************************/
+inline int32 CLuaBaseEntity::rotateToAngle(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isnumber(L, 1));
+
+    uint8 wantedRotation = (uint8)lua_tointeger(L, 1);
+
+    m_PBaseEntity->PAI->PathFind->RotateTo(wantedRotation);
+
+    return 0;
+}
+
+/************************************************************************
+* Function: pathStop()
+* Purpose : stops an npc from pathing until pathResume() is used
+* Example : npc:pathStop()
+* Notes   : used in conjunction with pathResume()
+************************************************************************/
+
+inline int32 CLuaBaseEntity::pathStop(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_PC);
+
+    CBattleEntity* PBattle = (CBattleEntity*)m_PBaseEntity;
+
+    PBattle->speed = 0;
+
+    return 1;
+}
+
+/************************************************************************
+* Function: pathResume()
+* Purpose : resumes an entity's pathing by looking at it's speed set in the DB 
+* Example : npc:pathResume()
+* Notes   : used in conjunction with pathStop()
+************************************************************************/
+
+inline int32 CLuaBaseEntity::pathResume(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_PC);
+
+    uint8 default_speed = 0;
+
+    if (m_PBaseEntity != nullptr)
+    {
+        if (m_PBaseEntity->objtype == TYPE_NPC)
+        {
+            auto PNpc = static_cast<CNpcEntity*>(m_PBaseEntity);
+            uint32 npcID = PNpc->id;
+            const char* NPCQuery = "SELECT speed FROM npc_list WHERE npcid = %u";
```

Comment:
Why not store the original speed in a local var instead of re-querying the database?

Reply:
For NPCs we could try having the event code itself stop and restart NPC movement. since `m_event.target` is the NPC entity itself. Mobs we'd still need a scritp command for stop/start.